operator/v1alpha1/types_kueue.go

```diff
@@ -0,0 +1,304 @@
+	ManageJobsWithoutQueueName *ManageJobsWithoutQueueNameOption `json:"manageJobsWithoutQueueName,omitempty"`
+	// managedJobsNamespaceSelector can be used to omit some namespaces from ManagedJobsWithoutQueueName
+	// Only valid if ManagedJobsWithoutQueueName is QueueNameOptional
+	// +kubebuilder:validation:XValidation:rule="self.manageJobsWithoutQueueName==QueueNameOptional",message="managedJobsNamespaceSelector is only valid if manageJobsWithoutQueueName is QueueNameOptional"
+	// +optional
+	ManagedJobsNamespaceSelector *metav1.LabelSelector `json:"managedJobsNamespaceSelector,omitempty"`
+	// fairSharing controls the fair sharing semantics across the cluster.
+	// +optional
+	FairSharing FairSharing `json:"fairSharing,omitempty"`
+}
+
+// KueueStatus defines the observed state of Kueue
+type KueueStatus struct {
+	operatorv1.OperatorStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// KueueList contains a list of Kueue
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type KueueList struct {
+	metav1.TypeMeta `json:",inline"`
+	// metadata for the list
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+	// items is a slice of kueue
+	// this is a cluster scoped resource and there can only be 1 kueue
+	// +kubebuilder:validation:MaxItems=1
+	// +required
+	Items []Kueue `json:"items"`
+}
+
+// These structs come directly from Kueue.
+type Resources struct {
+	// excludeResourcePrefixes defines which resources should be ignored by Kueue
+	// +optional
+	// +kubebuilder:validation:items:MaxLength=64
+	// +kubebuilder:validation:MaxItems=16
+	ExcludeResourcePrefixes []string `json:"excludeResourcePrefixes,omitempty"`
+
+	// transformations defines how to transform PodSpec resources into Workload resource requests.
+	// This is intended to be a map with Input as the key (enforced by validation code)
+	// +optional
+	// +kubebuilder:validation:MaxItems=16
+	Transformations []ResourceTransformation `json:"transformations,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=Retain;Replace
+type ResourceTransformationStrategy string
+
+// ResourceTransformations apply transformation to pod spec resources
+// Retain means that we will keep the original resources and
+// apply a transformation.
+// Replace means that the original resources will be replaced
+// after the transformation is done.
+const Retain ResourceTransformationStrategy = "Retain"
+const Replace ResourceTransformationStrategy = "Replace"
+
+// ResourceTransformations apply transformation to pod spec resources
+// Retain means that we will keep the original resources and
+// apply a transformation.
+// Replace means that the original resources will be replaced
+// after the transformation is done.
+type ResourceTransformation struct {
```

Comment:
I'm struggling to understand what this API actually does. Can you provide a concrete example of how I would configure this, and what the effect of that configuration would be?